test/rate_limiter_test.cc

```diff
@@ -39,4 +41,78 @@ TEST_F(RateLimiterTest, LinearRateLimiterInvalidArgumentTest) {
   ASSERT_DEATH(LinearRateLimiter rate_limiter(time_system, 0_Hz), "Frequency must be > 0");
 }
 
+TEST_F(RateLimiterTest, BurstingRateLimiterTest) {
+  const uint64_t burst_size = 3;
+  std::unique_ptr<MockRateLimiter> mock_rate_limiter = std::make_unique<MockRateLimiter>();
+  MockRateLimiter& unsafe_mock_rate_limiter = *mock_rate_limiter;
+  EXPECT_CALL(unsafe_mock_rate_limiter, tryAcquireOne)
+      .Times(burst_size)
+      .WillRepeatedly(Return(true))
+      .RetiresOnSaturation();
```

Comment:
What is `RetireOnSaturation`? Is this effectively the same as `InSequence`?

Reply:
Thanks, `InSequence` specifies the intent here more clearly and is more compact, so I switched it over to use that.